pkg/cmd/cli/secrets/basicauth.go

```diff
@@ -0,0 +1,219 @@
+type CreateBasicAuthSecretOptions struct {
+	SecretNamespace string
+	SecretName      string
+	Username        string
+	Password        string
+	CertificatePath string
+	GitConfigPath   string
+
+	PromptForPassword bool
+
+	Reader io.Reader
+	Out    io.Writer
+
+	SecretsInterface client.SecretsInterface
+}
+
+// NewCmdCreateBasicAuthSecret implements the OpenShift cli secrets new-basicauth subcommand
+func NewCmdCreateBasicAuthSecret(name, fullName string, f *kcmdutil.Factory, reader io.Reader, out io.Writer, newSecretFullName, ocEditFullName string) *cobra.Command {
+	o := &CreateBasicAuthSecretOptions{
+		Out:    out,
+		Reader: reader,
+	}
+
+	cmd := &cobra.Command{
+		Use:     fmt.Sprintf("%s SECRET_NAME --username=USERNAME --password=PASSWORD [--ca-cert=FILENAME --gitconfig=FILENAME]", name),
+		Short:   "Create a new secret for basic authentication",
+		Long:    createBasicAuthSecretLong,
+		Example: fmt.Sprintf(createBasicAuthSecretExample, fullName, newSecretFullName, ocEditFullName),
+		Run: func(c *cobra.Command, args []string) {
+			if err := o.Complete(f, args); err != nil {
+				kcmdutil.CheckErr(kcmdutil.UsageError(c, err.Error()))
+			}
+
+			if err := o.Validate(); err != nil {
+				kcmdutil.CheckErr(kcmdutil.UsageError(c, err.Error()))
+			}
+
+			if len(kcmdutil.GetFlagString(c, "output")) != 0 {
+				secret, err := o.MakeBasicAuthSecret()
+				kcmdutil.CheckErr(err)
+
+				kcmdutil.CheckErr(f.PrintObject(c, secret, out))
+				return
+			}
+
+			if err := o.CreateBasicAuthSecret(); err != nil {
+				kcmdutil.CheckErr(err)
+			}
+		},
+	}
+
+	cmd.Flags().StringVar(&o.Username, "username", "", "Username for Git authentication")
+	cmd.Flags().StringVar(&o.Password, "password", "", "Password or token for Git authentication")
+	cmd.Flags().StringVar(&o.CertificatePath, "ca-cert", "", "Path to the certificate file")
+	cmd.Flags().StringVar(&o.GitConfigPath, "gitconfig", "", "Path to the .gitconfig file")
+	cmd.Flags().BoolVarP(&o.PromptForPassword, "prompt", "", false, "Prompt for password or token")
+
+	// autocompletion hints
+	cmd.MarkFlagFilename("ca-cert")
+	cmd.MarkFlagFilename("gitconfig")
+
+	kcmdutil.AddPrinterFlags(cmd)
+
+	return cmd
+}
+
+// CreateBasicAuthSecret saves created Secret structure and prints the secret name to the output on success.
+func (o *CreateBasicAuthSecretOptions) CreateBasicAuthSecret() error {
+	secret, err := o.MakeBasicAuthSecret()
+	if err != nil {
+		return err
+	}
+
+	if _, err := o.SecretsInterface.Create(secret); err != nil {
+		return err
+	}
+
+	fmt.Fprintf(o.GetOut(), "secret/%s\n", secret.Name)
+	return nil
+}
+
+// MakeBasicAuthSecret builds up the Secret structure containing secret namespace, name, type and data structure
+// containing desired credentials.
+func (o *CreateBasicAuthSecretOptions) MakeBasicAuthSecret() (*api.Secret, error) {
+	secret := &api.Secret{}
+	secret.Namespace = o.SecretNamespace
```

Comment:
why are we setting namespace? won't that prevent them from using the generated secret with `oc create -f ... -n anynamespace`?